### PR TITLE
Use mod_floor when inserting into dict

### PR DESCRIPTION
**Pull request by Juan-M-V**
When creating a dictionary with negative values, the felts no longer remain negative. This fixes a previously untested bug that was found in cairo-rs-py

### CODE REVIEW

**Review by juanbono — APPROVED** [review]
LGTM

File: src/hint_processor/builtin_hint_processor/dict_manager.rs [hunk 1]
```diff
@@ -88,6 +90,12 @@ impl DictManager {
             ));
         };
 
+        for val in initial_dict.values_mut() {
+            if val.is_negative() {
+                *val = val.mod_floor(&vm.prime);
+            }
+        }
+
```
Comment (fmoletta) [inline src/hint_processor/builtin_hint_processor/dict_manager.rs:98]: Is it necessary to mutate the initial_dict we receive? Can't we mutate our copy of the dictionary, as we are already cloning it afterwards

Comment (Juan-M-V) [reply src/hint_processor/builtin_hint_processor/dict_manager.rs]: Resolved: https://github.com/lambdaclass/cairo-rs/pull/614/commits/9041c0cd93f048da6a252f3ec7aa0742d6321ffe

File: src/hint_processor/builtin_hint_processor/dict_manager.rs [hunk 2]
```diff
@@ -88,9 +90,19 @@ impl DictManager {
...
+        let mut floored_initial = HashMap::new();
+        for (key, val) in initial_dict.into_iter() {
+            let floored_val = if val.is_negative() {
+                val.mod_floor(&vm.prime)
+            } else {
+                val
+            };
+            floored_initial.insert(key, floored_val);
+        }
+
```
Comment (fmoletta) [inline src/hint_processor/builtin_hint_processor/dict_manager.rs:102]: Can we use map for this?

Comment (Juan-M-V) [reply src/hint_processor/builtin_hint_processor/dict_manager.rs]: Resolved: https://github.com/lambdaclass/cairo-rs/pull/614/commits/99220f47aac9f24a5dd0f20b533d8a512f66b6e9

**Review by Oppen — APPROVED** [review]

**Pull request merged**